### DISCUSSION TITLE
Bird Image Thumbnails?

### DISCUSSION THREAD

**Discussion by jkrauska** [tphakala/birdnet-go#331] [Q&A] [answered]
The README pics show bird thumbnails, but I'm not seeing that on docker install.

How are thumbnails used? (flickr api similar to birdnet-pi?)

**Comment by jkrauska** [comment]
figured out I need to use Dev version, but still not seeing some/many common bird images.
please advise?

<img width="1812" alt="Screenshot 2024-12-19 at 2 20 50 PM" src="https://github.com/user-attachments/assets/83380e87-46e2-466d-a489-5a9f284b4f44" />

**Reply by tphakala** [reply]
I think there are some bugs in thumbnail retrieval code as I have seen thumbnails going missing between birdnet-go restarts etc.

**Reply by tphakala** [reply]
I added a debug flag and debug prints for thumbnail retrieval and serving, https://github.com/tphakala/birdnet-go/commit/2ed97ffbcbc32756ce266667dd8f672c1ad2bebb

Update to latest dev image and enable debugging by setting thumbnails debug to true

```yaml
realtime:
   dashboard:
        thumbnails:
            debug: true
```

This should provide details why some thumbnails are missing

**Reply by tphakala** [reply]
I fixed some issues in thumbnail retrieval code, please test with latest code and check if you still see missing thumbnails

**Reply by jkrauska** [reply]
is there a way to detect which version/gitsha I'm running?

no images shown for "[Chestnut-backed Chickadee]" or "American Crow" which are very common birds.

will add the debug flag.

**Reply by jkrauska** [reply]
Author metadata missing:

```bash
2024/12/20 19:57:47 [2326bf52] Debug: Successfully retrieved Wikipedia page
2024/12/20 19:57:47 Debug: Processing image info response: 
```

```json
{
  "imageinfo": [
    {
      "extmetadata": {
        "Artist": {
          "source": "commons-desc-page",
          "value": "Louis Agassiz Fuertes (1874-1927) artist, authors of the written work variously listed as Harriman Alaska Expedition (1899), Edward Henry Harriman, Clinton Hart Merriam"
        },
        "Assessments": {
          "hidden": "",
          "source": "commons-categories",
          "value": ""
        },
        "AttributionRequired": {
          "hidden": "",
          "source": "commons-desc-page",
          "value": "false"
        },
        "Categories": {
          "hidden": "",
          "source": "commons-categories",
          "value": "CC-PD-Mark|Chickadees in art|Harriman Alaska Expedition|PD-Art (PD-old-70)|PD-Art (PD-old default)|PD US|PD art uploaded by Patrick Coin|Poecile rufescens (illustrations)"
        },
        "CommonsMetadataExtension": {
          "hidden": "",
          "source": "extension",
          "value": 1.2
        },
        "Copyrighted": {
          "hidden": "",
          "source": "commons-desc-page",
          "value": "False"
        },
        "Credit": {
          "source": "commons-desc-page",
          "value": "<i>Harriman Alaska series</i> (volume 1 or 2)"
        },
        "DateTime": {
          "hidden": "",
          "source": "mediawiki-metadata",
          "value": "2006-04-04 18:21:55"
        },
        "DateTimeOriginal": {
          "source": "commons-desc-page",
          "value": "1904"
        },
        "ImageDescription": {
          "source": "commons-desc-page",
          "value": "Chestnut-backed Chickadee, <i>Parus rufescens</i>, chromolithograph after painting (watercolor)"
        },
        "License": {
          "hidden": "",
          "source": "commons-templates",
          "value": "pd"
        },
        "LicenseShortName": {
          "hidden": "",
          "source": "commons-desc-page",
          "value": "Public domain"
        },
        "ObjectName": {
          "source": "mediawiki-metadata",
          "value": "Parus rufescensDU1N028CA"
        },
        "Permission": {
          "hidden": "",
          "source": "commons-desc-page",
          "value": "published in the United States pre-1923, also reproduction of art-work where artist died more than 70 years ago."
        },
        "Restrictions": {
          "hidden": "",
          "source": "commons-desc-page",
          "value": ""
        },
        "UsageTerms": {
          "source": "commons-desc-page",
          "value": "Public domain"
        }
      }
    }
  ],
  "imagerepository": "shared",
  "known": true,
  "missing": true,
  "ns": 6,
  "title": "File:Parus rufescensDU1N028CA.JPG"
}
```

```
2024/12/20 19:57:47 [2326bf52] Debug: Failed to fetch author info for Poecile rufescens: failed to get license URL from extmetadata: key not found
2024/12/20 19:57:47 Debug: Error fetching image for Poecile rufescens: failed to query thumbnail credit of bird: Poecile rufescens : failed to get license URL from extmetadata: key not found
```

**Reply by jkrauska** [reply]
2024/12/20 19:57:46 Debug: Error fetching image for Calypte anna: failed to query thumbnail credit of bird: Calypte anna : failed to get license URL from extmetadata: key not found

2024/12/20 19:57:47 Debug: Error fetching image for Poecile rufescens: failed to query thumbnail credit of bird: Poecile rufescens : failed to get license URL from extmetadata: key not found

**Reply by jkrauska** [reply]
can this just be made in to a pass with licenseURL="Uknown"?

https://github.com/tphakala/birdnet-go/blob/17f5cc873bb8bd0f1d695d73696b778fde60ef0e/internal/imageprovider/wikipedia.go#L287

**Comment by jkrauska** [comment]
@tphakala  now it's getting some more, but failing on missing artist metadata.

2024/12/21 13:38:17 Debug: Error fetching image for Corvus brachyrhynchos: failed to query thumbnail credit of bird: Corvus brachyrhynchos : failed to get artist from extmetadata: key not found

**Reply by isZumpo** [reply]
Originally the code was made to fail when missing author information. At the start the idea was to be really strict about copyright, always trying to give credits unless not needed by the license.

Wikimedia/wikimedia is a bit of a mess when it comes to the data, sometimes it exists and sometimes it does not. So to be on the safe side if the author was missing I made the decision to not show the image at all.

One idea I had at the time was that we could at some point in the future, for the edge cases like this do some manual work to ensure proper attribution.

**Reply by jkrauska** [reply]
I see active work to handle the messy cases.. thank you.

I think also that using a source with more standardized image sizes might be a better route.


eg
https://www.allaboutbirds.org/news/search/?q=american+crow
https://www.allaboutbirds.org/guide/assets/photo/59858041-240px.jpg

would we be open to alternative image providers?

**Reply by isZumpo** [reply]
That website contains a lot of information & details. Nice!

From a technical standpoint, adding additional image providers is not that hard. We only have to be sure that our use of any third-party service aligns with its terms of use. In addition some of the APIs, for instance, the [eBird API](https://documenter.getpostman.com/view/664302/S1ENwy59) which I assume allaboutbirds uses in the background, require an API key for certain requests (for instance taxonomy info). Requiring API keys is fine, but it makes it a bit more difficult for everyone to use Birdnet-Go since it would be one additional thing to configure.

In addition to images, I would like to see us add in some bird information page, where one would be able to read about the detected birds. Something similar to what allaboutbirds offers :)

**Comment by jkrauska** [comment, marked as answer]
all birds in my detections now are rendering:

ghcr.io/tphakala/birdnet-go   dev       66ff6484728f   3 hours ago    609MB
